src/bin/nydus-image/core/chunk_dict.rs

```diff
@@ -100,9 +103,48 @@ impl HashChunkDict {
             blob_idx_m: Mutex::new(BTreeMap::new()),
         };
 
+        let config = target_rafs_config.unwrap_or_else(|| rs.meta.get_config());
```

Comment:
Following code can be abstract as RafsConfig::check_compatibility(&rs.meta)?

Reply:
Ok, thanks.